src/types/ResourceManager.ts

```diff
@@ -0,0 +1,30 @@
+import { Context, RESOURCE_CACHE_KEY } from './Context';
+import Resource from '../resources/Resource';
+
+export default class ResourceManager {
```

Comment:
```suggestion
export default class ResourceCache {
```

or `ResourceCacheManager`?